### ISSUE TITLE
how to train with more than 3 channels

### ISSUE THREAD

**Issue by magistri** [facebookresearch/maskrcnn-benchmark#768]
## ❓ Questions and Help
Does anyone know how to change the input shape of the network? I'd like to use 4 or 5 channels images.

Thanks.

**Comment by kia69028** [comment]
Simply add more channels when building up the backbone

**Comment by zimenglan-sysu-512** [comment]
thanks

**Comment by Antoine-ls** [comment]
@magistri I have the same question, have you solved it?

**Comment by magistri** [comment]
yes! I have added a parameters in config/defaults.py to specify the input channels which is then read while building the backbone. For example in resnet.py you can change  conv1 in the BaseStem class.

**Comment by Antoine-ls** [comment]
@magistri COOL! Thanks. By the way, have you changed the dataloader to load multi-channel images?

**Comment by magistri** [comment]
I haven't cause I combined multiple images in the dataset file.

**Comment by Antoine-ls** [comment]
@magistri  So you write your own dataset file instead of coco format or PASCAL format? Can you please tell what files to change cause I have problems creating my own dataset with multi-channels. Thanks!

**Comment by magistri** [comment]
Instead of using `img, anno = super(COCODataset, self).__getitem__(idx)` in the `__getitem__` function I load the image according to my own problem structure. You can also keep the coco.py structure and overwrite the img variable

**Comment by Antoine-ls** [comment]
@magistri oh,yeah,I got it. But when I implement it, I meet a problem: the img variable in the coco.py is an 'Image' object, but when I concatenate two images into one array, and use Image.fromarray to get an Image object, there would be a problem as Image object may only have 3 channels? if I just keep numpy array as return type for img, there would be other errors, in my case, it happens in #851 . Could you please tell me your return data type for img ? Thanks in advance!

**Comment by magistri** [comment]
I load the image with PIL, then `img1 = img1.numpy(), img2 = img2.numpy(), img = np.concatenate((img1, img2), axis=0) `

**Comment by prafulag** [comment]
Does the image has to be PIL format?
I am trying to use it for some medical images and they are 32bit float range. However, PIL does not seem to support 3 channel for 32bit float images. Any thoughts ?

**Comment by prafulag** [comment]
@magistri 
Does that mean the __getitem__ function can return a tensor of say HxWx3 (or whatever number of input channels), in place of a PIL Image object?

**Comment by Antoine-ls** [comment]
@prafulag No matter what format, it is transform function which turns PIL format into tensor, so you can load your image as a numpy array instead of PIL format and use ToTensor to turn it into tensor for return

**Comment by prafulag** [comment]
@Antoine-ls 
Thanks for the quick response. I am infact returning it as a torch tensor.

**Comment by cskarthik7** [comment]
I did the same thing as magistri did but I get nan loss! Because it forces me to eliminate the pretrained backbone model so I dont know how to go about it!
